### ISSUE TITLE
Ruby Ranges highlight incorrectly 

### ISSUE THREAD

**Issue by joshuabowers** [jneen/rouge#375]
The first period of the range is styled correctly as a punctuation character; the second is styled as a Name.Function (e.g. `.nf`).

For example, the following code creates for ranges of different type.

``` ruby
integer_range = 0..10
float_range = 0.1..0.9
string_range = 'a'..'z'

exclusive_range = 0...10
```

The first three are inclusive, so the `..` should be treated as a single punctuation entity. The fourth creates a range between 0 and 9; in this case, `...` is a single operator.

See, e.g., this output:
![screen shot 2016-01-24 at 6 46 38 pm](https://cloud.githubusercontent.com/assets/275265/12541410/a048a646-c2cb-11e5-98ea-7e12a981412e.png)

**Comment by kjcpaas** [comment]
In relation to this, the decimal point in floats have a different color. The `.` is also not being treated as part of the number in this case. Am I correct?

<img width="482" alt="screen shot 2016-11-25 at 11 36 32 pm" src="https://cloud.githubusercontent.com/assets/3772828/20629746/124a92a8-b368-11e6-967d-754c94ddf64b.png">

I added these cases in `spec/visual/ruby` so the color difference is more obvious. When I compared it to the syntax highlighter in my editor, `2.3` has 1 color so it's treated as 1 object.

<img width="60" alt="screen shot 2016-11-25 at 11 40 11 pm" src="https://cloud.githubusercontent.com/assets/3772828/20629872/a88756d4-b368-11e6-91f4-6532e7f099a1.png">

I'd like to take a crack on this one @jneen :bowing_woman:
